core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -566,22 +587,45 @@ class SpendReportingService(
     traceFutureWithParent("getSpendForAllWorkspaces", ctx) { childContext =>
       validateReportParameters(start, end)
       for {
+        // get spend-reportable workspaces, grouped by their spend export config
+        // Map[BillingProjectSpendExport, Seq[Workspace]]
         billingMap <- getBillingWithSpendPermission(childContext)
         _ = if (billingMap.isEmpty) {
-          return Future.successful(None)
-        }
-        projectNames: Map[GoogleProjectId, WorkspaceName] = billingMap.values.flatten.toMap
-        tableToProjectIdsMap: Map[String, Seq[GoogleProjectId]] = billingMap.map { case (table, workspaces) =>
-          table -> workspaces.map(_._1)
+          throw new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.NotFound, "No workspaces eligible for spend report")
+          )
```

Comment:
this now throws instead of early-returning `Future.successful(None)`. Akka was hanging trying to send `None` in the response payload; now the API returns properly in cases where the user has no workspaces on which to report. The UI should avoid this case but it was possible via the API.